lib/public/views/QcFlags/QcFlagsModel.js

```diff
@@ -103,4 +104,36 @@ export class QcFlagsModel extends Observable {
     get forSimulationPassOverviewModel() {
         return this._forSimulationPassOverviewModel;
     }
+
+    /**
+     * Load the QC flag details model
+     *
+     * @returns {void}
+     */
+    loadDetails({ id }) {
+        this._detailsModel = new QcFlagDetailsModel(
+            { id: parseInt(id, 10) },
+            () => {
+                /**
+                 * Workaround, as there is no information about QC flag association to data/simulation pass,
+                 * cannot simple go to proper overview page when flag is deleted
+                 */
+                if (this.model.router.history.length >= 3) {
+                    this.model.router.history.back();
+                } else {
+                    this.model.router.go('/?page=home');
+                }
```

Comment:
Why 3? This can't work, and is a very ugly hack...
What you can do however, is simply to pass from details, the data/simulation pass id and runs and stuff?

Reply:
refactored, added missing parameters to url